### ISSUE TITLE
Encoding Object examples have invalid schema property

### ISSUE THREAD

**Issue by jeremyfiel** [OAI/OpenAPI-Specification#3170]
Reviewing the [Encoding Object Examples](https://spec.openapis.org/oas/v3.0.3#special-considerations-for-multipart-content)

The keyword `type` should be `$ref`. 

```yaml
         addresses:
            # default Content-Type for arrays is based on the `inner` type (object shown, so `application/json` in this example)
            type: array
            items:
              type: '#/components/schemas/Address'
```

**Comment by jeremyfiel** [comment]
I noticed [3.1.0](https://spec.openapis.org/oas/v3.1.0#special-considerations-for-multipart-content) has a slightly different schema, albeit, valid.  Lmk if you want that updated too. 

```yaml
          addresses:
            # default Content-Type for arrays is based on the _inner_ type (object shown, so `application/json` in this example)
            type: array
            items:
              type: object
              $ref: '#/components/schemas/Address'
```

**Comment by handrews** [comment]
@jeremyfiel I'm not sure what would need to be fixed in the 3.1 schema?  I might be missing something.

**Comment by jeremyfiel** [comment]
the `type: object` is redundant with a `$ref` sibling

**Comment by handrews** [comment]
@jeremyfiel I suspect I put it in there just to emphasize that you can put things next to `$ref` now :-)  I don't much care one way or the other, but I think that's why it's like that.

**Comment by jeremyfiel** [comment]
i'm impartial.

**Comment by handrews** [comment]
Let's leave it as-is and just fix it for 3.0.4.

**Comment by handrews** [comment]
This has been fixed in 3.0.4 and we decided not to change it in 3.1.1.
